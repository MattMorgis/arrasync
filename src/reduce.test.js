const {reduce} = require('./reduce');

describe('Reduce Tests', () => {
  test('Reduces into single value', async () => {
    const list = [1, 2, 3, 4, 5];
    const result = await reduce(
      list,
      (item, accumlator) => {
        return item + accumlator;
      },
      0
    );
    expect(result).toBe(15);
  });

  test('Reduce uses optional arguments', async () => {
    const testArray = ['a', 'b', 'c', 'd'];
    const result = await reduce(
      testArray,
      (accumulator, value, i, array) => {
        accumulator.push([value, i, array[0]]);
        return accumulator;
      },
      []
    );
    expect(result).toMatchSnapshot();
  });
});
