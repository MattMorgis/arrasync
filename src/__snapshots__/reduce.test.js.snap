// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reduce Tests Reduce  1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce u 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce us 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce use 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses  1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses o 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses op 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses opti 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optio 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses option 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optiona 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional  1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional a 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional ar 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional argu 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional argum 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional argume 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional argument 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduce uses optional arguments 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduces  1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduces 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduces into single val 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduces into single valu 1`] = `
Array [
  Array [
    "a",
    0,
    "a",
  ],
  Array [
    "b",
    1,
    "a",
  ],
  Array [
    "c",
    2,
    "a",
  ],
  Array [
    "d",
    3,
    "a",
  ],
]
`;

exports[`Reduce Tests Reduces into single value 1`] = `[Function]`;
